packages/fluentui/react-builder/src/components/NarrationComputer.tsx

```diff
@@ -0,0 +1,268 @@
+    // First, handle some special case conditions
+    let value;
+    if (definitionName === 'textarea') {
+      // Begin if 1
+      value = (element as HTMLTextAreaElement).value.trim();
+    } // End if 1
+    if (definitionName === 'textarea' && platform === 'Win/JAWS' && value) {
+      // Begin if 1
+      this.computedParts.description = SRNC.stateStrings['Win/JAWS']['textarea']['[containsText]'];
+    } else {
+      // else if 1
+      this.computedParts.description =
+        element
+          .getAttribute('aria-describedby')
+          ?.split(/\s+/)
+          .map(x => {
+            return element.ownerDocument?.getElementById(x)?.textContent || null;
+          })
+          .filter(x => x !== null)
+          .join('') || '';
+    } // End if 1
+  } // End computeDescription
+
+  // Computes and stores the accessible name and title parts of the narration for the given element using the given computed node.
+  computeNameAndTitle(node: any, element: Element) {
+    this.computedParts.name = node.name;
+
+    // If the title attribute is present, set its value as the description part of the narration if it was not computed as accessible name and if no accessible description was computed before
+    let title = element.getAttribute('title');
+    if (title && this.computedParts.name !== title && !this.computedParts.description) {
+      // Begin if 1
+      this.computedParts.description = title;
+    } // End if 1
+  } //End computeNameAndTitle
+
+  // Computes the value part of the narration using the given computed node.
+  computeValue(node: any) {
+    this.computedParts.value = node.valueText;
+  } //End computeValue
+
+  // Computes the position part of the narration for the given definitionName.
+  computePosition(definitionName: string) {
+    // The computation of the position in set would be too difficult, so use just "X of Y" instead
+    const positionRoles = ['role=menuitem', 'role=menuitemcheckbox', 'role=menuitemradio', 'role=tab', 'role=option'];
+    this.computedParts.position = positionRoles.includes(definitionName) ? '[X of Y]' : '';
+  } // End computePosition
+
+  // Computes the type and state parts of the narration for the given definitionName, element and platform using the given computed node.
+  computeTypeAndState(node: any, definitionName: string, element: Element, platform: string) {
+    // Set the default ttype and state for unknown roles and element types
+    this.computedParts.type = `[${node.role}]`;
+    this.computedParts.state = '';
+
+    // Find the rule which matches the states that are present on the element
+    const rules = SRNC.stateRules[platform][definitionName];
+    if (rules) {
+      // Begin if 1
+      rulesLoop: for (let i = 0; i < rules.length; i++) {
+        // Begin for 1
+        const rule = rules[i];
+        const possibleStates = SRNC.possibleStates[definitionName];
+        for (let j = 0; j < possibleStates.length; j++) {
```

Comment:
I feel like this could be a `possibleStates.forEach`?

Reply:
As mentioned above, I have used `Array.some()` instead of `Array.forEach()`